### ISSUE TITLE
`Cursor` causes crashes if the texture is moved out of bounds

### ISSUE THREAD

**Issue by kirides** [Lehona/LeGo#33]
if `_Cursor_GetVal` calls `View_MoveToPxl()` with `coordinates + cursor size (64px)`  > `Print_Screen[PS_Y]` the games performance goes to a crawl and sooner or later crashes.

i propose a check for cursor texture width & height.
For example User providable through userconstants as users might want to use a smaller cursor.TGA, and benefit the higher "precision" of not having a 64x64 texture.
I tested this with a 34x34 cursor, which felt more "natural" as you could almost move the cursor to the bottom line of the screen. Basically at the same height of the Bars.

**Comment by Lehona** [comment]
I'm not quite sure I understand - `_Cursor_GetVal` does bounds-checking/clamping to make sure the View is never moved outside the screen. Do you mean the case where the texture is partially outside the screen, but the actual pointer is still inside? Does it happen on the horizontal axis too?

**Comment by kirides** [comment]
Yes. GetVal does not check including the texture size, only the XY pos.This causes a hangup if the player were to move the cursor to, let's say x= 8150, as the textures width requires additional 64px.The issue is seen if you open a zSpy, the call to View_MoveToPxl causes it.

**Comment by kirides** [comment]
Funnily enough, the game _sometimes_ freezes, _sometimes_ disables the mouse controls and _sometimes_ just works, with the cursor going "OOB" with it's texture. That's on a clean installation. with and without zSpy.

This warning is spammed into the zSpy
```
Warn:  0 X:     [RND3D]SetViewport: Set viewport failed! Error: DDERR_INVALIDPARAMS .... <zRndD3D.h,#127>
Warn:  0 C:     d3dviewport values: dwX: 875, dwY: 1377, dwWidth:  64, dwHeight: 61, dvMinZ: 0, dvMaxZ: 1 .... <zRndD3D_Render.cpp,#1952>
```
